File: app/src/main/java/br/com/pedrosilva/tecnonutri/domain/repositories/FeedRepository.java
package br.com.pedrosilva.tecnonutri.domain.repositories;

import java.util.List;

import br.com.pedrosilva.tecnonutri.domain.entities.FeedItem;

/**
 * Created by psilva on 3/16/17.
 */

public interface FeedRepository {

    void get(String feedHash, FeedItemCallback callback);

    void likeItem(String feedHash);

    void dislikeItem(String feedHash);

    boolean isLiked(String feedHash);

    void firstList(FeedCallback callback);

    void loadMore(int page, int timestamp, FeedCallback callback);

    interface FeedCallback {
        void onSuccess(List<FeedItem> feedItems, int page, int timestamp);
        void onError(Throwable t);
    }

    interface FeedItemCallback {
        void onSuccess(FeedItem feedItem);
        void onError(Throwable t);
    }
}

File: app/src/main/kotlin/br/com/pedrosilva/tecnonutri/domain/repositories/FeedRepository.kt
package br.com.pedrosilva.tecnonutri.domain.repositories

import br.com.pedrosilva.tecnonutri.domain.entities.FeedItem

interface FeedRepository {

    fun get(feedHash: String, callback: FeedItemCallback)

    fun likeItem(feedHash: String)

    fun dislikeItem(feedHash: String)

    fun isLiked(feedHash: String): Boolean

    fun firstList(callback: FeedCallback)

    fun loadMore(page: Int, timestamp: Int, callback: FeedCallback)

    interface FeedCallback {

        fun onSuccess(
            feedItems: List<FeedItem>,
            page: Int,
            timestamp: Int
        )

        fun onError(error: Throwable)
    }

    interface FeedItemCallback {

        fun onSuccess(feedItem: FeedItem)

        fun onError(error: Throwable)
    }
}